pkg/auth/dependency/interaction/flows/webapp.go

```diff
@@ -212,7 +240,7 @@ func (f *WebAppFlow) TriggerOOBOTP(token string, step interaction.Step) (*WebApp
 		panic("interaction_flow_webapp: unexpected interaction state")
 	}
 
-	err = f.Interactions.PerformAction(i, step, &interaction.ActionTriggerOOBAuthenticator{
+	err = f.Interactions.PerformAction(i, s.Steps[0].Step, &interaction.ActionTriggerOOBAuthenticator{
```

Comment:
I think it should be using `s.CurrentStep()` instead of `s.Steps[0]`?

Reply:
Actually I would like to know by design, when would `getState` return more than 1 step?

If there are more than 1 step, the first step in the steps slice should be the next step the user should take, right? `CurrentStep()` return the last step in the steps slice.

Another question I encountered is that some of the code is now checking `len(s.Steps) != 1`, which may not be true in the future because `getState` might include more steps. But as long as the next step is expected, we should not panic.

Reply:
> Actually I would like to know by design, when would getState return more than 1 step?

`s.Steps` is supposed to be a list of executed steps + current step to execute. For example in a login interaction, it derives the steps like this:
1. If need primary authentication, add 'authenticate primary' to step list
2. If need primary authentication, and not yet performed it, return the current step list
3. If need secondary authentication, add 'authenticate secondary' to step list
4. If need secondary authentication, and not yet performed it, return the current step list
5. Add 'commit' to step list and return

So possible step list are:
- authenticate primary
- authenticate primary, authenticate secondary
- authenticate primary, authenticate secondary, commit

The last step in step list reflect `s.CurrentStep()`, i.e. the step that the flow should execute.

> If there are more than 1 step, the first step in the steps slice should be the next step the user should take, right? CurrentStep() return the last step in the steps slice.

The first step in the step list is the first step that the flow should execute. It may (length > 1) or may not (length = 1) been executed.

> Another question I encountered is that some of the code is now checking len(s.Steps) != 1, which may not be true in the future because getState might include more steps. But as long as the next step is expected, we should not panic.

These checks are mostly redundant, and should be replaced with `s.CurrentStep()`, since these checks occurs at the beginning of interaction, it is expected that there is exactly 1 step in the list.